### ISSUE TITLE
(CodePipelines): consider renaming "Release Change" button

### ISSUE THREAD

**Issue by warden** [aws/aws-cdk#14697]
When working with CodePipelines I find the "Release Change" button quite confusing, in many cases I would tend to say that it's doing the opposite of releasing a change. An example being:

- If a changeset failed to apply, the pipeline is not retried. The pipeline must be restarted manually from the top by clicking Release Change.
- A stack that failed to deploy must be deleted manually using the CloudFormation console before starting the pipeline again by clicking Release Change.

Reference: https://docs.aws.amazon.com/cdk/latest/guide/cdk_pipeline.html

In both cases this means the button should be named "restart pipeline", with probably a warning note appearing in (2) that the user needs to manually delete the CF stack first.

This goes back to the Concepts defined [here](https://docs.aws.amazon.com/codepipeline/latest/userguide/concepts.html) where it's stated:
_An execution is a set of changes released by a pipeline. Each pipeline execution is unique and has its own ID. An execution corresponds to a set of changes, such as a merged commit or a manual release of the latest commit._

I personally find this definition wrong, there is no _release_ happening here, it's a _deployment_ or a _build_. 
In other words, usually one commit is not a release (unless every commit goes to production which is a special case). 
The word released can be substituted here by "applied" or "executed". The button on the pipeline is another story and is just a "Trigger" for a pipeline, so it could be named "Start Pipeline" or "Restart" or "Re-run" or whatever else is applicable to the current state of the pipeline.

By definition of a CD system, there is a difference between a pipeline and a release. A pipeline is a series of execution steps, a release is a set of features, potentially connected with a version(number, name). For me mixing those two is totally  confusing.

Comments are welcome :)

**Comment by skinny85** [comment]
Hi @warden,

thanks for opening the issue. Is there anything CDK-related here, or are you just taking the opportunity of opening an issue to the CodePipeline team through this repository? 🙂

Thanks,
Adam

**Comment by warden** [comment]
Hi @skinny85,

Not really, it's more nomenclature related. I opened it here as I felt as the best place to go from the links in "Did this page help you?" footer. If you know a better place to send this to, let me know and I'll move it there, thanks :)

**Comment by skinny85** [comment]
What page did you start at, and pressed the "Did this page help you?" button, that made you arrive here?

**Comment by warden** [comment]
[The one that is referred in the description](https://docs.aws.amazon.com/cdk/latest/guide/cdk_pipeline.html), open that one and ctrl+f exactly for the text from the first bullet-point.

**Comment by skinny85** [comment]
OK, I see. We won't be able to do anything about the CodePipeline UI, but we are able to change the wording in that document.

Tagging in @Jerry-AWS, who's our doc writer, and owns that page.

**Comment by Jerry-AWS** [comment]
I don't think there's anything to change in the Guide. I'll see about getting this to the right people in CodePipeline.

**Comment by peterwoodworth** [comment]
@Jerry-AWS have you sent this to the CodePipeline people?

**Comment by Jerry-AWS** [comment]
Reopening to remind me.
